### PR TITLE
Feature: query string API endpoint

### PR DESCRIPTION
**Pull request by Mogball**
Exposes an endpoint `superset/slices/<slice_id>/query` that returns the corresponding query string and language for that slice in the form `{ query: querystring, language: querylang }`, which would be seen when clicking on `Query` in the explore slice view.

Also includes some unit tests.

### CODE REVIEW

**Comment by mistercrunch** [comment]
There's a bit of duplication of logic with what is going on here: https://github.com/apache/incubator-superset/blob/master/superset/views/core.py#L972 . What is the use-case for this endpoint?

Perhaps we can add corresponding routing for `slice_id` querystring instead of `form_data`. I think we do it with redirect logic in some places, may work for you, or we could avoid redirecting and just serve from that url.

One behavior that we want to have/preserve is around being able to reference a slice and override some elements easily.

**Comment by Mogball** [comment]
@mistercrunch is it alright if I just move the logic into a separate function? Redirecting to `/explore_json/<datasource_type>/<datasource_id>` from `/slice/<slice_id>/query/` would mean calling `get_viz` a second time, which makes another database call.

**Comment by Mogball** [comment]
As for the use case: we want to be able to plug the generated query into other systems.

File: superset/views/core.py [hunk 1]
```diff
@@ -2312,6 +2315,20 @@ def sqllab(self):
             entry='sqllab',
             bootstrap_data=json.dumps(d, default=utils.json_iso_dttm_ser)
         )
+
+    @api
+    @has_access_api
+    @expose("/slice/<slice_id>/query/")
+    def sliceQuery(self, slice_id):
```
Comment (mistercrunch) [inline superset/views/core.py]: NIT: `slice_query`

**Pull request merged**